core/nodepool/config/templates/stack-template.json

```diff
@@ -349,7 +349,7 @@
                   "Resource": "arn:{{.Region.Partition}}:s3:::{{ .KubeResourcesAutosave.S3Path }}/*"
                 },
                 {{end}}
-                {{if or .Kube2IamSupport.Enabled .KIAMSupport.Enabled }}
+                {{if .Kube2IamSupport.Enabled }}
```

Comment:
Ah, yes. We have a nodeSelector to stick kiam server to controller nodes. So we don't need to consult `.KIAMSupport.Enabled` here. Good catch!